src/source/parserutil.d

```diff
@@ -27,7 +27,17 @@ auto unexpectedTokenError(Lexer)(ref Lexer lexer, string expected) {
 		: format!"expected %s, not `%s`."(expected,
 		                                  token.toString(lexer.context));
 
+	const reachedEOF = lexer.index >= lexer.content.length - 1;
```

Comment:
Can't you call the reachedOEF function directly?

Reply:
Totally forgot about that

Reply:
`JsonLexer` doesn't have `reachedEOF` though

Reply:
Are you sure? jsonlexer mixin LexBaseImpl, which mixin LexBaseUtils, so you should have it.

Reply:
`LexBaseUtils` is added to private section, so `unexpectedTokenError` doesn't see it

Reply:
Ok, I think we can move LexBaseUtils int he public section, if we tighten the visibility within LexBaseUtils itself.